### PR TITLE
SSCS-8496 Process reasonable adjustment event callback

### PR DESCRIPTION
**Pull request by jackmaloney**
https://tools.hmcts.net/jira/browse/SSCS-8496

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processreasonableadjustment/ProcessReasonableAdjustmentAboutToSubmitHandler.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> callbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        if (sscsCaseData.getReasonableAdjustmentsLetters() == null || sscsCaseData.getReasonableAdjustmentsLetters().size() == 0) {
+            callbackResponse.addError("No reasonable adjustment correspondence has been generated on this case");
+        }
+
+        checkReasonableAdjustmentsOutstandingFlag(sscsCaseData);
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processreasonableadjustment/ProcessReasonableAdjustmentAboutToSubmitHandler.java:40]: Can call sscsCaseData.updateReasonableAdjustmentsOutstanding() instead

File: build.gradle [hunk 2]
```diff
@@ -307,7 +307,7 @@ dependencies {
     compile group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
     compile group: 'uk.gov.hmcts.reform', name: 'doc-assembly-client', version: '1.0.6'
     compile group: 'uk.gov.hmcts.reform', name: 'document-management-client', version: '7.0.0'
-    compile group: 'uk.gov.hmcts.reform', name: 'sscs-common', version: '4.1.33'
+    compile group: 'uk.gov.hmcts.reform', name: 'sscs-common', version: 'DEV-SNAPSHOT'
```
Comment (ryandac) [inline build.gradle:310]: Can use version 4.1.35.RC-SSCS-8495-1

Comment (ryandac) [reply build.gradle:310]: should also keep sscs-pdf-email-common to 1.4.20-RC-SSCS-8495-1

**Review by ryandac — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (jackmaloney) [reply build.gradle:310]: Yeh that was intentional as wanted to wait until 8495 was merged. I've now rebased my branch and created a new  4.1.35.RC-SSCS-8495-3 version

**Review by ryandac — APPROVED** [review]

**Pull request merged**